### PR TITLE
caggs on joins

### PR DESCRIPTION
**Pull request by Loquacity**
# Description

Adds section about using caggs on JOINed tables to create a cagg section.

# Links

Fixes https://github.com/timescale/timescaledb/pull/4874

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/timescaledb/latest/contribute-to-docs/)

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

### CODE REVIEW

File: timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md [hunk 1]
```diff
@@ -142,14 +142,31 @@ queries to run efficiently.
 
 </procedure>
 
+## Create a continuous aggregate on JOINed tables
+
```
Comment (RafiaSabih) [inline timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md:146]: I am finding this term JOINed tables a bit odd/unclear, excuse me if it is a general term used by customers or user community. I would have preferred following wording for this, perform joins in the definition of continuous aggregates, or perform joins while creating/defining continuous aggregates, something like that.

**Comment by RafiaSabih** [comment]
Somewhere it has to be mentioned that performing joins with using clause is not supported for postgresql 12 version, but only for later versions.

File: timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md (hunk 1 above)
Comment (charislam) [reply timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md:146]: Agreed, I think I'd prefer `Create a continuous aggregate with a JOIN in the definition`

File: timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md [hunk 2]
```diff
@@ -142,14 +142,31 @@ queries to run efficiently.
...
+In TimescaleDB 2.9 and later, with PostgreSQL 13 or later, you can create a
+continuous aggregate on tables that are connected with a `JOIN`. For example:
```
Comment (charislam) [inline timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md:148]:
```suggestion
continuous aggregate from a query that uses a `JOIN`. For example:
```

Is this better? The underlying table itself is not permanently connected with a `JOIN`, it's the query (and now the continuous aggregate) that is, so maybe the wording is a bit less confusing? Will bow to the expertise of someone who knows SQL better though

**Review by charislam — APPROVED** [review]

File: timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md (hunk 1 above)
Comment (Loquacity) [reply timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md:146]: I fiddled with the wording a bit. Is it clearer now?

**Comment by Loquacity** [comment]
> Somewhere it has to be mentioned that performing joins with using clause is not supported for postgresql 12 version, but only for later versions.

I wrote "In TimescaleDB 2.9 and later, with PostgreSQL 13 or later". Does that cover it?

**Comment by Loquacity** [comment]
@RafiaSabih Are you happy for this to merge now?

**Comment by RafiaSabih** [comment]
> > Somewhere it has to be mentioned that performing joins with using clause is not supported for postgresql 12 version, but only for later versions.
> 
> I wrote "In TimescaleDB 2.9 and later, with PostgreSQL 13 or later". Does that cover it?

Not really, because we can perform joins in PG 12 as well just not with the USING clause. So this requires a mention.

**Review by RafiaSabih — APPROVED** [review]

**Comment by iroussos** [comment]
@Loquacity @RafiaSabih we need to add more info on the joins in continuous aggregates:
1. Joins in continuous aggregates is supported starting at 2.10.0, not 2.9
2. Most of our pages still mention that we do not support joins in caggs
  - The canonical API page for continuous aggregates: https://docs.timescale.com/api/latest/continuous-aggregates/create_materialized_view/
  - The troubleshooting page for continuous aggregates - https://docs.timescale.com/timescaledb/latest/how-to-guides/continuous-aggregates/troubleshooting/#queries-fail-when-defining-continuous-aggregates-but-work-on-regular-tables
  - Not sure if more pages have wrong references of not supporting joins but at least the main pages for caggs should mention that we support joins and link to a reference page with more information
3. The [page we added ](https://docs.timescale.com/timescaledb/latest/how-to-guides/continuous-aggregates/create-a-continuous-aggregate/#create-a-continuous-aggregate-with-a-join) has no further information, the restrictions, what is allowed etc. We should at least mention on a SSOT section/page about those restrictions, like for example that only joining with regular tables is allowed, only inner joins, etc
4. (Nice to have) It would be great to also update the table with the supported features on https://docs.timescale.com/timescaledb/latest/how-to-guides/continuous-aggregates/about-continuous-aggregates/

This blocks us from officially announcing the 2.10.0 release - no proper documentation is equal to not having the feature out and this is the major feature released with2.10.0. I would really appreciate if we can prioritize making those updates. Thank you!

/cc @horzsolt

**Comment by RafiaSabih** [comment]
The new PR for docs changes --> https://github.com/timescale/docs/pull/2089
@iroussos 

> (Nice to have) It would be great to also update the table with the supported features on https://docs.timescale.com/timescaledb/latest/how-to-guides/continuous-aggregates/about-continuous-aggregates/
is more complicated to add since this table only have 2.6 and below and 2.7 and above, I am not sure if we should have a third column for 2.10 or new set of columns for 2.9 and below and 2.10 and above. So I am leaving that for now.

**Pull request merged**